src/ui/common/src/handlers/responses/dag.ts

```diff
@@ -0,0 +1,42 @@
+// This file should mirror src/golang/workflow/dag/response.go
+import { EngineConfig } from '../../utils/engine';
+import { ExecutionStatus } from '../../utils/shared';
+import { StorageConfig } from '../../utils/storage';
+import {
+  WorkflowRetentionPolicy,
+  WorkflowSchedule,
+} from '../../utils/workflows';
+import { ArtifactResponse, ArtifactResultResponse } from './artifact';
+import { OperatorResponse, OperatorResultResponse } from './operator';
+
+export type DagMetadataResponse = {
+  dag_id: string;
+  dag_created_at: number;
+  storage_config?: StorageConfig;
+  engine_config?: EngineConfig;
+
+  workflow_id: string;
+  workflow_created_at: number;
+  user_id: string;
+  name: string;
+  description: string;
+  schedule?: WorkflowSchedule;
+  retention_policy?: WorkflowRetentionPolicy;
+};
+
+export type DagResponse = DagMetadataResponse & {
+  operators: { [id: string]: OperatorResponse };
+  artifacts: { [id: string]: ArtifactResponse };
+};
+
+export type DagRawResultResponse = {
```

Comment:
I wonder if there is a more descriptive name for this result. DagRawResultResponse doesn't let me know what it contains. I think DagMetadatResponse would be more descriptive but that name is already in use.

Reply:
yeah I had a hard time figuring this out as well. This is technically a `ResultResponse` but that's also used by the main `ResultResponse`

Reply:
Maybe use something like DagResultStatus here @likawind ?